src/asmjs/shared-constants.cpp

```diff
@@ -67,6 +67,8 @@ cashew::IString GLOBAL("global"),
                 ALMOST_ASM("almost asm"),
                 BUFFER("buffer"),
                 ENV("env"),
+                STACKTOP("STACKTOP"),
+                STACK_MAX("STACK_MAX"),
```

Comment:
If I was feeling picky I would ask you split this into three separate PR:

1. Moving/using of shared constants
2. Fix the ctor eval issue
3. Import STACKTOP.

Its up to you if you actaully want to do that.

Reply:
Fair point, these are separable things. I split this into 3 commits as suggested, and can land this with rebase-and-merge.